### ISSUE TITLE
pip install -e . seems to break with poetry-core 1.7.0

### ISSUE THREAD

**Issue by jreniel** [mtkennerly/poetry-dynamic-versioning#139]
poetry-core release 1.7.0 states:

- Remove deprecated poetry.core.constraints.generic and poetry.core.semver (https://github.com/python-poetry/poetry-core/pull/601).
https://github.com/python-poetry/poetry-core/releases/tag/1.7.0

When using `poetry-core 1.7.0` and using `pip install .` or `pip install -e .`, we get:
```
pip._vendor.pyproject_hooks._impl.BackendUnavailable: Traceback (most recent call last):                                                                                                   
  File "/home/jreniel/.cache/pypoetry/virtualenvs/pyschism-hi2If_f8-py3.10/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 77, in _build_backend
    obj = import_module(mod_path)                                                                                                                                                          
  File "/home/jreniel/.pyenv/versions/3.10.12/lib/python3.10/importlib/__init__.py", line 126, in import_module                                                                            
    return _bootstrap._gcd_import(name[level:], package, level)                                                                                                                            
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import                                                                                                                          
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load                                                                                                                       
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked                                                                                                              
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked                                                                                                                        
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module                                                                                                                  
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed                                                                                                             
  File "/tmp/pip-build-env-biqnf1x7/overlay/lib/python3.10/site-packages/poetry_dynamic_versioning/backend.py", line 5, in <module>                                                        
    patch.activate()
  File "/tmp/pip-build-env-biqnf1x7/overlay/lib/python3.10/site-packages/poetry_dynamic_versioning/patch.py", line 52, in activate
    _apply_patches()
  File "/tmp/pip-build-env-biqnf1x7/overlay/lib/python3.10/site-packages/poetry_dynamic_versioning/patch.py", line 43, in _apply_patches
    _patch_poetry_create(factory_mod)
  File "/tmp/pip-build-env-biqnf1x7/overlay/lib/python3.10/site-packages/poetry_dynamic_versioning/patch.py", line 14, in _patch_poetry_create
    from poetry.core.semver.version import Version as PoetryVersion
ModuleNotFoundError: No module named 'poetry.core.semver'
```

As a workaround I used:
`requires = ["poetry-core>=1.0.0,<1.7.0", "poetry-dynamic-versioning"]`
and it worked.

**Comment by olimcc** [comment]
We also faced a manifestation of this error, receiving the following error when referencing a local dependency: 
```
ChefBuildError

Backend 'poetry_dynamic_versioning.backend' is not available.

at /usr/local/lib/python3.10/site-packages/poetry/installation/chef.py:147 in _prepare
    143│
    144│                 error = ChefBuildError("\n\n".join(message_parts))
    145│
    146│             if error is not None:
  → 147│                 raise error from None
    148│
    149│             return path
    150│
    151│     def _prepare_sdist(self, archive: Path, destination: Path | None = None) -> Path:

```
Modifying our `build-system` configuration as guided by @jreniel resolved the issue for us for now:
```
[build-system]
requires = ["poetry-core>=1.0.0,<1.7.0", "poetry-dynamic-versioning"]
build-backend = "poetry_dynamic_versioning.backend"
```
Prior to this, we had no upper bound on `poetry-core`.

**Comment by mtkennerly** [comment]
Thanks for reporting this :+1: I'm working on a fix now.

**Comment by mtkennerly** [comment]
This is now fixed in [v1.0.1](https://github.com/mtkennerly/poetry-dynamic-versioning/releases/tag/v1.0.1).

**Comment by jreniel** [comment]
Thanks @mtkennerly for your awesomeness and hard work, which keeps our packages lean, mean and going!
